### PR TITLE
Kokspflanze multiple formatter

### PR DESCRIPTION
**Pull request by ThaDafinser**
Rebase of #174

### CODE REVIEW

**Comment by ThaDafinser** [comment]
@kokspflanze please review. If all is ok i will merge it

**Comment by kokspflanze** [comment]
i will check that, this evening

File: docs/03. Columns.md [hunk 1]
```diff
@@ -435,13 +435,13 @@ $imageFormatter->setAttribute('alt', 'Image Alt');
...
-$col->setFormatter($imageFormatter);
+$col->setFormatters($imageFormatter);
 ```
 
 ### Link
-The Link formatter displays a column content as an HTML link with value and href is the column content, to use it do the following:
+The Link formatters displays a column content as an HTML link with value and href is the column content, to use it do the following:
 ```php
-$col->setFormatter(new Formatter\Link());
+$col->setFormatters(new Formatter\Link());
```
Comment (kokspflanze) [inline docs/03. Columns.md]: that must be an array so like `setFormatter([new Formatter\Link()])`

Comment (ThaDafinser) [inline docs/03. Columns.md]: Thank you. fixed

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 2]
```diff
@@ -602,39 +613,81 @@ public function getRendererParameters($rendererName = 'jqGrid')
...
+    /**
+     * Set a template formatter and overwrite other formatter
      *
      * @param AbstractFormatter $formatter
+     * @deprecated please use setFormatters
      */
     public function setFormatter(AbstractFormatter $formatter)
     {
-        $this->formatter = $formatter;
+        trigger_error('Please use setFormatters()', E_DEPRECATED);
```
Comment (kokspflanze) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: E_DEPRECATED
will generate in ErrorLog:
PHP Warning:  Invalid error type specified in ......ZfcDatagrid/Column/AbstractColumn.php on line 633

i think its better to use
E_USER_DEPRECATED
will generate in ErrorLog:
PHP Deprecated:  Please use setFormatters() in 

check also http://php.net/manual/en/errorfunc.constants.php

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 3]
```diff
@@ -602,39 +613,81 @@ public function getRendererParameters($rendererName = 'jqGrid')
...
         }
 
         return false;
     }
 
+    public function hasFormatter()
+    {
+        trigger_error('Please use hasFormatters()', E_DEPRECATED);
+
+        return $this->hasFormatter();
```
Comment (kokspflanze) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: hasFormatters =)

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 4]
```diff
@@ -602,39 +613,81 @@ public function getRendererParameters($rendererName = 'jqGrid')
...
     /**
      *
-     * @param  string  $rendererType
      * @return boolean
      */
-    public function hasFormatter()
+    public function hasFormatters()
     {
-        if ($this->formatter !== null) {
+        if (count($this->formatters) > 0) {
```
Comment (kokspflanze) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: a empty or bool cast would be faster as a count

**Comment by kokspflanze** [comment]
my question is atm you deleted many times `return $this` that will be a big BC break, i think you want a little clean up for 1.0.0 or?

check done =)

**Comment by ThaDafinser** [comment]
@kokspflanze yes it's a cleanup for 1.0. Read the reason here: http://ocramius.github.io/blog/fluent-interfaces-are-evil/

**Pull request merged**